modules/components/pages/inputs/nats_stream.adoc

```diff
@@ -78,7 +79,7 @@ The NATS Streaming Server is being deprecated. Critical bug fixes and security f
 
 Tracking and persisting offsets through a durable name is also optional and works with or without a queue. If a durable name is not provided then subjects are consumed from the most recently published message.
 
-When a consumer closes its connection it unsubscribes, when all consumers of a durable queue do this the offsets are deleted. In order to avoid this you can stop the consumers from unsubscribing by setting the field `unsubscribe_on_close` to `false`.
+When a consumer closes its connection it unsubscribes. When all consumers of a durable queue do this the offsets are deleted. To stop consumers from unsubscribing, set the field `unsubscribe_on_close` to `false`.
```

Comment:
```suggestion
When a consumer closes its connection, it unsubscribes from the subject. When all consumers of a durable queue do this, the offsets are deleted. To stop consumers from unsubscribing, set the field `unsubscribe_on_close` to `false`.
```
Could we clarify what the consumers unsubscribe from? And are the offsets deleted in the NATS server, or somewhere else?

Reply:
@mihaitodor. Can you answer Kat's question about when a consumer unsubscribes?

Reply:
Unfortunately I'm not super-familiar with the Jetstream internals, but it has some mechanisms to keep shared state consistent in the Jetstream server. So when a client connects to it and reads data, it then sends ack messages back to acknowledge that a message was read. Then the server is responsible to somehow preserve that acknowledgement by incrementing the corresponding offset.

Not sure what is the question about unsubscribing. The way I understand `unsubscribe_on_close` based on the original docs is that if all consumers of a given subject disconnect and they have `unsubscribe_on_close: false` in the config, then even if all of them disconnect, the offsets are persisted so they can reconnect later and resume where they left off. If `unsubscribe_on_close: true`, then my guess is that as soon as the last one disconnects, then the server is expected to delete that offset state, so if the consumers reconnect later, they'll have to start from scratch. I haven't tested this, so I could be wrong.

Reply:
Thanks Mihai. You answered the question. I'll try and confirm that with Ash.

Reply:
@Jeffail, Just wanted to check that the changes I've just made are accurate. Could you take a look?

Reply:
Yep LGTM